wildstore-meta/src/main/java/edu/sjsu/wildstore/meta/controller/ShareLinkController.java

```diff
@@ -212,15 +214,22 @@ public long getShareLinkCount(OAuth2AuthenticationToken oAuth2AuthenticationToke
     @DeleteMapping("/{shareId}")
     public boolean deleteShareLink(@PathVariable String shareId) {
         try {
-            Query query = new Query(Criteria.where("shareId").is(shareId));
-            mongoTemplate.remove(query, DBObject.class, "share-links");
+            Query query = new Query(Criteria.where("_id").is(shareId));
+            mongoTemplate.remove(query, SHARE_LINKS_COLLECTION);
             return true;
         } catch (Exception ex) {
             System.out.println(ex.getMessage());
             return false;
         }
     }
 
+    @PreAuthorize("hasRole('USER')")
```

Comment:
we should make this ADMIN for now. if we do user, we need to make sure that users can only delete their own links.